src/main/java/com/google/devtools/build/lib/bazel/bzlmod/ModuleExtensionId.java

```diff
@@ -80,7 +80,7 @@ public static ModuleExtensionId create(
   }
 
   public boolean isInnate() {
-    return extensionName().contains("%");
```

Comment:
Why do we need this change?

Reply:
This is for the third point in the description: module extension IDs with an isolation key and those for repo rules would conflict after serialization into a string (for the lockfile), which results in a crash when parsing them back.

Reply:
this seems dangerous to me. Changing the separator to `+` here would mean that (IIUC) we can't have it in the .bzl file label, so essentially using any repo rule from a non-main repo would fail (because that would result in an extension name like `@@foo++foo_ext+foo_repo//:bar.bzl+my_repo_rule`). In fact I'm surprised all tests passed -- maybe we never `use_repo_rule` from non-main repos in our tests?

(and yes, I understand that this also means `%` was never allowed to show up in the .bzl file label, but this was also true of aspects etc.)

Reply:
Hmm, why exactly would this fail? The synthetic extension name as passed in by module file evaluation is "prettified" to just `_repo_rules` for the purpose of generating the extension unique prefix: https://cs.opensource.google/bazel/bazel/+/master:src/main/java/com/google/devtools/build/lib/bazel/bzlmod/BazelDepGraphFunction.java;l=186?q=_repo_rules&ss=bazel%2Fbazel&start=41.

Reply:
Yes, the repos generated by an innate extension have the form of `module+_repo_rulesN+repo_name`. But the name of the extension corresponding to the repo name prefix `module+_repo_rulesN` is currently `<bzl_file_label>%<rule_name>` ([code](https://source.corp.google.com/piper///depot/google3/third_party/bazel/src/main/java/com/google/devtools/build/lib/bazel/bzlmod/InnateRunnableExtension.java;rcl=698515353;l=107)). This PR is changing that line to split on `+`, which _should_ result in errors.

Reply:
I'm now using a space instead of a plus and am also only splitting on the last occurrence of the separator.

I don't think this was an issue in practice though as the bzl file label is the one specified in a module file and thus uses apparent rather than canonical repo names (I don't think we ever intentionally supported the latter and even prohibited them at some point).

Reply:
ah, I see! That's indeed why everything was actually okay.

Yeah, as long as the separator is not a valid Starlark identifier character and we split by the last occurrence only, we should be all good in all cases. This LGTM now, thanks!